tests/unit/components/form/HdRange.spec.js

```diff
@@ -0,0 +1,99 @@
+import { wrapperFactoryBuilder } from 'tests/unit/helpers';
+import HdRange from '@/components/form/HdRange.vue';
+
+describe('HdRange', () => {
+  let wrapper;
+  const initialValue = 25;
+  const maxValue = 100;
+  const minValue = 0;
+  const wrapperFactory = wrapperFactoryBuilder(HdRange, {
+    propsData: {
+      name: 'testRange',
+      minValue,
+      maxValue,
+      value: initialValue,
+    },
+  });
+
+  beforeEach(() => {
+    wrapper = wrapperFactory();
+  });
+
+  test('renders component', () => {
+    expect(wrapper.html()).toMatchSnapshot();
+  });
+
+  test('renders with preselected value', () => {
+    expect(parseFloat(wrapper.find('input').element.value)).toBe(initialValue);
+  });
+
+  test('respects the maximum value passed', () => {
+    wrapper.setProps({
+      value: maxValue * 2,
+    });
+
+    expect(parseFloat(wrapper.find('input').element.value)).toBe(maxValue);
+    expect(wrapper.emitted('input')[1][0]).toEqual(maxValue);
+  });
+
+  test('limits the value to the passed minimum value', () => {
+    wrapper.setProps({
+      value: minValue - 100,
+    });
+
+    expect(parseFloat(wrapper.find('input').element.value)).toBe(minValue);
+    expect(wrapper.emitted('input')[1][0]).toEqual(minValue);
+  });
+
+  test('When the focus event is emitted, the proper method is executed', () => {
+    const mockedFocusHandler = jest.fn();
+    wrapper.setMethods({ focusHandler: mockedFocusHandler });
+
+    wrapper.find('input').trigger('focus');
+
+    expect(mockedFocusHandler).toHaveBeenCalledTimes(1);
+  });
+
+  test('When the focus event is emitted, the input is in an active status', () => {
+    wrapper.find('input').trigger('focus');
+
+    expect(wrapper.vm.isActive).toBe(true);
```

Comment:
I think those two guys can also be inside the same test (if I'm missing some context, please let me know).

Reply:
I can not put them together. The _isActive_ value is set by the _focusHandler_ method. In the first test, I'm mocking that method, so the _isActive_ value is not affected. So, I can make just one test from this two, but I would, however, have to trigger _focus_ twice. So, it's pointless to have just one test.

Reply:
How about not mocking it on the first test?
You can just spy the `focusHandler` so you can test it is being called and the `isActive` state is set as well.
What do you think?

Reply:
I changed the order and it works good!